tests/conftest.py

```diff
@@ -1330,8 +1330,33 @@ def pytest_generate_tests(metafunc):        # noqa E302
 
     if "enum_dut_portname" in metafunc.fixturenames:
         metafunc.parametrize("enum_dut_portname", generate_port_lists(metafunc, "all_ports"))
-    if "enum_dut_portname_module_fixture" in metafunc.fixturenames:
-        metafunc.parametrize("enum_dut_portname_module_fixture", parametrise_autoneg_tests(), scope="module")
+
+    if "enum_dut_portname_module_fixture" in metafunc.fixturenames or \
+            "enum_speed_per_dutport_fixture" in metafunc.fixturenames:
+        autoneg_tests_data = get_autoneg_tests_data()
+        if "enum_dut_portname_module_fixture" in metafunc.fixturenames:
+            def format_id(param):
+                return "{}|{}|{}".format(param['dutname'], param['port'], ','.join(param['speeds']))
+            metafunc.parametrize(
+                "enum_dut_portname_module_fixture",
+                autoneg_tests_data,
+                scope="module",
+                ids=format_id,
+                indirect=True
+            )
+
+        if "enum_speed_per_dutport_fixture" in metafunc.fixturenames:
+            def format_id(param):
```

Comment:
@antonptashnik Why keep two same functions here? Is it better to move it out and define a common one?